### ISSUE TITLE
 After loading the JVM library, the Library instance was not retained, causing methods like get_env to fail to execute properly.

### ISSUE THREAD

**Issue by Kyle-Derrick** [jni-rs/jni-rs#550]
After loading the JVM library, the Library instance was not retained, causing methods like get_env to fail to execute properly.
![image](https://github.com/user-attachments/assets/f5ffefc8-24b1-49f6-becf-34518cfd2e9a)

## Env

os: windows 10
rust: v1.82.0
jni: v0.21.1

## Code

```rust
// ...
let jvm = JavaVM::new(init_args).unwrap();
let mut env = jvm.get_env().unwrap();    // Error msg: called `Result::unwrap()` on an `Err` value: JniCall(ThreadDetached)
// ...
```

> After testing and verification, retaining the Library value can resolve the issue.

**Comment by argv-minus-one** [comment]
The `ThreadDetached` error occurs because you didn't attach the current thread to the JVM after starting it. You need to call `JavaVM::attach_current_thread` to do that.

See, for example, how one of the tests does it: https://github.com/jni-rs/jni-rs/blob/9278710b5d8a580f24d4b06c02ff7fb86b0821a9/tests/invocation_character_encoding.rs#L23-L25

The JVM library (`libjvm.so`, `libjvm.dylib`, or `jvm.dll`) hasn't been unloaded. If it had been, your program would crash from trying to call a function (namely [`GetEnv`](https://docs.oracle.com/en/java/javase/11/docs/specs/jni/invocation.html#getenv)) in a library that's no longer loaded, not raise a `ThreadDetached` error.

---

Although now that you mention it, I'm surprised that the program *doesn't* crash, because you're right that dropping the `Library` should cause the JVM library to be unloaded. Not only would this cause a crash when you try to call any JNI function, this should also instantly crash all of the threads that HotSpot automatically creates.

It would seem that HotSpot is somehow not being unloaded while it's running. [I tried finding how it's preventing that](https://github.com/search?q=repo%3Aopenjdk%2Fjdk+dlopen&type=code&p=1), but didn't find anything.

At any rate, it would probably be best to prevent the `Library` from being dropped. I've submitted PR #554.

**Comment by Kyle-Derrick** [comment]
Yes, this is an issue that is easily overlooked. It is important to ensure that the lifecycle of `libjvm` is consistent with `JavaVM`, as this may lead to some unexpected problems. Moreover, `attach_current_thread` does not actually attach to the main thread, which could result in discrepancies from what the user expects, such as issues with the absence of the current thread's associated ClassLoader.

**Comment by argv-minus-one** [comment]
> It is important to ensure that the lifecycle of `libjvm` is consistent with `JavaVM`

Has to exceed it, actually. Dropping a `JavaVM` does not terminate the JVM.

> Moreover, `attach_current_thread` does not actually attach to the main thread

I'm afraid I don't understand. As far as I know, it does. What am I missing?
